backend/routes/room_routes.py

```
@@ -121,6 +122,54 @@ def get_room(room_id):
+        # Validate input data
+        if "name" in data and not data["name"].strip():
+            return jsonify({"error": "Room name cannot be empty"}), 400
+        if "capacity" in data:
+            try:
+                capacity = int(data["capacity"])
+                if capacity <= 0:
+                    return jsonify({"error": "Capacity must be a positive number"}), 400
+            except ValueError:
+                return jsonify({"error": "Invalid capacity value"}), 400
+        if "id_structure" in data:
+            structure = db.query(Structure).filter(Structure.id == data["id_structure"]).first()
```

Ruff (0.8.2)

Undefined name `Structure`

(F821)